cmd/api/src/database/sso_providers.go

```diff
@@ -43,3 +44,28 @@ func (s *BloodhoundDB) CreateSSOProvider(ctx context.Context, name string, authP
 
 	return provider, CheckError(s.db.WithContext(ctx).Table(ssoProviderTableName).Create(&provider))
 }
+
+func (s *BloodhoundDB) GetAllSSOProviders(ctx context.Context, order string, sqlFilter model.SQLFilter) ([]model.SSOProvider, error) {
+	var providers []model.SSOProvider
+
+	query := s.db.WithContext(ctx).Model(&model.SSOProvider{})
+
+	// Apply SQL filter if provided
+	if sqlFilter.SQLString != "" {
+		query = query.Where(sqlFilter.SQLString, sqlFilter.Params...)
+	}
+
+	// Apply sorting order if provided
+	if order != "" {
+		query = query.Order(order)
+	} else {
+		// Default ordering by created_at if no order is specified
+		query = query.Order("created_at")
+	}
+
+	// Preload the associated OIDC and SAML providers
+	query = query.Preload("OIDCProvider").Preload("SAMLProvider")
```

Comment:
I am generally against gorm and it'll be fun to tease this out later, but I will concede this is neat!

Reply:
Yup, credit to Dillon for sending the helpful GORM doc above!